src/contracts/tokens/ProtocolToken.sol

```diff
@@ -33,4 +36,14 @@ contract ProtocolToken is ERC20Votes, Authorizable, IProtocolToken {
   function burn(uint256 _wad) external {
     _burn(msg.sender, _wad);
   }
+
+  // --- Overrides ---
+
+  function _update(address _from, address _to, uint256 _value) internal override(ERC20, ERC20Votes) {
+    super._update(_from, _to, _value);
+  }
+
+  function nonces(address _owner) public view override(ERC20Permit, IERC20Permit, Nonces) returns (uint256 _nonce) {
+    return super.nonces(_owner);
+  }
```

Comment:
is this because of importing ERC20Permit And ERC20Votes? doesn't it import it by default? 
i believe this could be avoided with another tree of imports

Reply:
@0xJabberwock 👀

Reply:
> is this because of importing ERC20Permit And ERC20Votes?

It is because of inheriting `ERC20Permit` and `ERC20Votes`.

> doesn't it import it by default?

`ERC20Votes` does not import `ERC20Permit` by default since v5.0.0.

From [OpenZeppelin's changelog](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/CHANGELOG.md):
> `ERC20Votes`: Changed internal vote accounting to reusable `Votes` module previously used by `ERC721Votes`. Removed implicit `ERC20Permit` inheritance. Note that the `DOMAIN_SEPARATOR` getter was previously guaranteed to be available for `ERC20Votes` contracts, but is no longer available unless `ERC20Permit` is explicitly used; ERC-5267 support is included in `ERC20Votes` with `EIP712` and is recommended as an alternative. (https://github.com/OpenZeppelin/openzeppelin-contracts/pull/3816)

Reply:
> i believe this could be avoided with another tree of imports

I tried different inheritance patterns, but it seems that overriding within the inheriting contract is not avoidable while it `is ERC20Permit, ERC20Votes`.

In addition, even the inheritance of the general interface (in this case, `IProtocolToken`) brings issues as it `is IERC20Permit`, when `ERC20Permit` contract already `is IERC20Permit`.

😵‍💫

Reply:
damn, thanks for the research 🚀